source/Halibut/Transport/Proxy/HttpProxyClient.cs

```diff
@@ -337,9 +340,7 @@ void SendConnectionCommand(string host, int port)
         
         async Task SendConnectionCommandAsync(string host, int port, CancellationToken cancellationToken)
         {
-            // TODO - ASYNC ME UP!
-            // This stream needs to be wrapped into a TimeoutStream
-            var stream = TcpClient.GetStream();
+            var stream = new NetworkTimeoutStream(TcpClient.GetStream());
```

Comment:
Did you have extension methods on `TcpClient` to make a timed out stream?

Reply:
No, we could / we can just do this simple wrap!!